core/src/traits.rs

```diff
@@ -74,3 +81,45 @@ pub trait Func<Input> {
     /// use cases.
     fn call(i: Input) -> Self::Output;
 }
+
+/// This is a user-implementable alternative to `FnMut`.
+///
+/// Not necessary if/when the FnMut trait is implementable in stable Rust.
+///
+/// ```
+/// # #[macro_use]
+/// # extern crate frunk;
+/// # use frunk::{FuncMut, PolyMut};
+/// # use std::fmt::{Debug, Write};
+///
+/// struct ConcatDebug<'a>(&'a mut String);
+///
+/// impl<'a, T: Debug> FuncMut<T> for ConcatDebug<'a> {
+///     type Output = ();
+///     fn call(&mut self, t: T) {
+///         write!(self.0, "{:?}", t).unwrap();
+///     }
+/// }
+///
+/// fn main() {
```

Comment:
Hide the `fn main() {`.

There's some examples of how I would format an example like this in
https://github.com/lloydmeta/frunk/blob/master/src/validated.rs